lib/cargo.js

```diff
@@ -163,33 +170,114 @@ export function provideBuilder() {
         });
       }
 
+      // Shows panic info
+      function showPanic(panic) {
+        // Only add link if we have panic.filePath, otherwise it's an external link
+        atom.notifications.addError(
+          'A thread panicked at '
+              + (panic.filePath ? '<a id="' + panic.id + '" href="#">' : '')
+              + 'line ' + panic.line + ' in ' + panic.file
+              + (panic.filePath ? '</a>' : ''), {
+                detail: panic.message,
+                stack: panic.stack,
+                dismissable: true
+              });
+        if (panic.filePath) {
+          const link = document.getElementById(panic.id);
+          if (link) {
+            link.panic = panic;
+            link.addEventListener('click', function (e) {
+              atom.workspace.open(e.target.panic.filePath, {
+                searchAllPanes: true,
+                initialLine: e.target.panic.line - 1
+              });
+            });
+          }
+        }
+      }
+
+      // Tries to parse a stack trace. Returns the quantity of actually parsed lines.
+      function tryParseStackTrace(lines, i, panic) {
+        let parsedQty = 0;
+        let line = lines[i];
+        if (line.substring(0, 16) === 'stack backtrace:') {
```

Comment:
ah, fine with me then, leave it as is, we can always go around adjusting stuff later